src/network/network_coordinator.cpp

```diff
@@ -208,8 +240,8 @@ bool ClientNetworkCoordinatorSocketHandler::Receive_SERVER_CONNECTING(Packet *p)
 	assert(connecter_it != this->connecter_pre.end());
 
 	/* Now store it based on the token. */
-	this->connecter_pre.erase(connecter_it);
 	this->connecter[token] = connecter_it->second;
+	this->connecter_pre.erase(connecter_it);
```

Comment:
In commit 'Add: support STUN for network connections' you move this line by one, should it be put in the right place in the commit that added these lines?